### PR TITLE
[BUGFIX] {{filter-by}} must work without prototype extensions

### PR DESCRIPTION
**Pull request by cibernox**

### CODE REVIEW

File: tests/integration/helpers/find-by-test.js [hunk 1]
```diff
@@ -24,6 +24,22 @@ test('It finds a value by a property', function(assert) {
...
 
+test('It finds a value by a property in arrays without prototype extensions', function(assert) {
+  this.set('array', [
+    { foo: true, name: 'a' },
+    { foo: false, name: 'b' },
+    { foo: true, name: 'c' }
+  ]);
+
+  this.render(hbs`
+    {{~#with (find-by 'name' 'b' array) as |item|~}}
```
Comment (martndemus) [inline tests/integration/helpers/find-by-test.js]: Use double-quotes in HBS

Comment (martndemus) [inline tests/integration/helpers/find-by-test.js]: Ugh, the whole file is this way :(
I'll fix in separate PR.

File: addon/helpers/find-by.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ export default Helper.extend({
       let array = get(this, 'array');
       let value = get(this, 'value');
 
-      return array.findBy(byPath, value);
+      return array.find(e => get(e, byPath) === value);
```
Comment (martndemus) [inline addon/helpers/find-by.js]: I think it's preferable to wrap `array` in `emberArray` and then just use `findBy`.

Comment (cibernox) [inline addon/helpers/find-by.js]: Out of curiosity, why is that preferable? (Unless this addons targets 1.13 and ie8)

Comment (martndemus) [inline addon/helpers/find-by.js]: Because IMO a `findBy` reads better then a terse arrow function.

**Comment by martndemus** [comment]
Fixed by #195

**Pull request closed without merging**